00 - Notes/ทีซีพี-ไอพี, ไอพีแอสเดรส.md

```diff
@@ -1,9 +1,7 @@
-ภาษา คือสิ่งที่ใช้แสดงออกแทนความคิด เป็นหัวใจหลักของการสื่อสาร ใช้เพื่อให้คู่สนทนาเข้าใจในสิ่งที่ต้องการสื่อสาร ผู้ส่งสารและผู้รับสาร จำเป็นต้องสื่อสารในภาษาที่ทั้งสองเข้าใจ การสื่อสารถึงจะมีประสิทธิมากที่สุด ปัจจุบันมีหลากหลายภาษา แต่ละภาษาใช้ในกลุ่มคนที่ต่างกันตามประเทศที่ตนเองอาศัย เช่น คนไทยใช้ภาษาไทย และคนจีนก็ใช้ภาษาจีน และถ้าเมื่อคนไทยต้องสื่อสารกับคนจีน จำเป็นต้องใช้ภาษาที่ทั้งสองฝั่งเข้าใจ มนุษย์เลยยกให้ภาษาอังกฤษ เป็นภาษาสากลของโลก แต่ภาษาไม่ได้ถูกใช้แค่กับมนุษย์ ยังมีภาษาสำหรับคอมพิวเตอร์ที่ใช้ในการสื่อสารกันอีกด้วย โดยภาษาที่คอมพิวเตอร์ใช้สื่อสารกันนั้นเรียกว่า TCP/IP
+ภาษา คือสิ่งที่ใช้แสดงออกแทนความคิด เป็นหัวใจหลักของการสื่อสาร ใช้เพื่อให้คู่สนทนาเข้าใจในสิ่งที่ตนเองต้องการจะสื่อสาร ดังนั้นผู้ส่งสารและผู้รับสารจำเป็นต้องสื่อสารในภาษาที่ทั้งสองเข้าใจ การสื่อสารถึงจะมีประสิทธิมากที่สุด ปัจจุบันมีหลากหลายภาษา แต่ละภาษาใช้ในกลุ่มคนที่ต่างกันตามประเทศที่ตนเองอาศัย เช่น คนไทยใช้ภาษาไทย และคนจีนก็ใช้ภาษาจีน และถ้าเมื่อคนไทยต้องสื่อสารกับคนจีน จำเป็นต้องใช้ภาษาที่ทั้งสองฝั่งเข้าใจ มนุษย์เลยยกให้ภาษาอังกฤษ เป็นภาษาสากลของโลก แต่ภาษาไม่ได้ถูกใช้แค่กับมนุษย์ ยังมีภาษาสำหรับคอมพิวเตอร์ที่ใช้ในการสื่อสารกันอีกด้วย โดยภาษาที่คอมพิวเตอร์ใช้สื่อสารกัน จะมีทั้งหมด 2 ภาษา คือ TCP/IP และ UDP แต่โดยส่วนใหญ่แล้วจะนิยมใช้ TCP/IP
```

Comment:
UDP, TCP layer เดียวกัน, IPข้างล้าง

Reply:
ไม่ใช่มีแค่ 2 ภาษาที่คอมพิวเตอร์ใช้สื่อสารกันในระบบ network นะครับ เรียกว่าเป็นที่นิยมหรือคุ้นหู จะมี 2 protocols (tcp and udp) จะเหมาะสมกว่าครับ แต่โดยส่วนใหญ่ที่เราเห็นๆ กันจะนิยมใช้ TCP 

ref: https://en.wikipedia.org/wiki/Lists_of_network_protocols
https://sonicautomation.co.th/%E0%B8%84%E0%B8%A7%E0%B8%B2%E0%B8%A1%E0%B9%81%E0%B8%95%E0%B8%81%E0%B8%95%E0%B9%88%E0%B8%B2%E0%B8%87%E0%B8%A3%E0%B8%B0%E0%B8%AB%E0%B8%A7%E0%B9%88%E0%B8%B2%E0%B8%87-tcp-%E0%B8%81%E0%B8%B1%E0%B8%9A-udp/#:~:text=TCP%20%E0%B8%A2%E0%B9%88%E0%B8%AD%E0%B8%A1%E0%B8%B2%E0%B8%88%E0%B8%B2%E0%B8%81%20Transmission,%E0%B8%81%E0%B8%B2%E0%B8%A3%E0%B9%80%E0%B8%8A%E0%B8%B7%E0%B9%88%E0%B8%AD%E0%B8%A1%E0%B8%95%E0%B9%88%E0%B8%AD%E0%B8%A3%E0%B8%B0%E0%B8%AB%E0%B8%A7%E0%B9%88%E0%B8%B2%E0%B8%87%E0%B8%AA%E0%B8%AD%E0%B8%87